x-pack/legacy/plugins/spaces/server/lib/copy_to_spaces/lib/rereadable_stream.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { Transform, TransformCallback, Readable } from 'stream';
+
+export class Rereadable extends Transform {
+  private chunks: any[] = [];
+
+  constructor() {
+    super({ writableObjectMode: true, readableObjectMode: true });
+  }
+
+  _transform(chunk: any, encoding: string, callback: TransformCallback) {
+    this.chunks.push(_.cloneDeep(chunk));
```

Comment:
Deep clone is required because the [`bulkCreate` procedure](https://github.com/elastic/kibana/blob/e137477b5a00b8c83e63acda1121c1d2ee10541c/src/core/server/saved_objects/service/lib/repository.ts#L205) used by `import` mutates the objects from the stream, and each "reread" of the stream needs to be pure.

Reply:
I'll see if I can track this down again, but IIRC, it had something to do with the `namespace` attribute. The copy operation would succeed for the first space in the request, but wouldn't end up copying anything for the other spaces in the request.